internal/api/courses/assignments/submissions/proxy/resubmit.go

```diff
@@ -0,0 +1,99 @@
+package proxy
+
+import (
+	"context"
+
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/db"
+	"github.com/edulinq/autograder/internal/grader"
+	"github.com/edulinq/autograder/internal/model"
+	"github.com/edulinq/autograder/internal/timestamp"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+type ResubmitRequest struct {
+	core.APIRequestAssignmentContext
+	core.MinCourseRoleGrader
+	TargetSubmission string `json:"target-submission"`
+
+	ProxyUser core.TargetCourseUser `json:"proxy-email"`
+	ProxyTime *timestamp.Timestamp  `json:"proxy-time"`
+
+	Message string `json:"message"`
+}
+
+func (request ResubmitRequest) GetContext() context.Context {
+	return request.Context
+}
+
+func (request ResubmitRequest) GetAssignment() *model.Assignment {
+	return request.Assignment
+}
+
+func (request ResubmitRequest) GetUserEmail() string {
+	return request.ProxyUser.Email
+}
+
+func (request ResubmitRequest) GetMessage() string {
+	return request.Message
+}
+
+func (request ResubmitRequest) ToLogAttrs() []any {
+	return core.GetLogAttributesFromAPIRequest(&request)
+}
+
+type ResubmitResponse struct {
+	FoundUser       bool `json:"found-user"`
+	FoundSubmission bool `json:"found-submission"`
+
+	core.BaseSubmitResponse
```

Comment:
We usually put embedded types first.
(Not always, but it's usually nice in smaller structs like this.)